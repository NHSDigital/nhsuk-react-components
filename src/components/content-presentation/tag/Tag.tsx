import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';

interface TagProps extends HTMLProps<HTMLSpanElement> {
  color?:
    | 'white'
    | 'grey'
    | 'green'
    | 'aqua-green'
    | 'blue'
    | 'purple'
    | 'pink'
    | 'red'
    | 'orange'
    | 'yellow';
}

const Tag: FC<TagProps> = ({ className, color, ...rest }) => (
  <strong
    className={classNames('nhsuk-tag', { [`nhsuk-tag--${color}`]: color }, className)}
    {...rest}
  />
);

export default Tag;
