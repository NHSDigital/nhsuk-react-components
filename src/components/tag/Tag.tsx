import React, { HTMLProps } from 'react';
import classNames from 'classnames';

type TagColours =
  | 'white'
  | 'grey'
  | 'green'
  | 'aqua-green'
  | 'blue'
  | 'purple'
  | 'pink'
  | 'red'
  | 'orange'
  | 'yellow';

export interface TagProps extends HTMLProps<HTMLSpanElement> {
  color?: TagColours;
}

const Tag: React.FC<TagProps> = ({ className, color, ...rest }) => (
  <span
    className={classNames('nhsuk-tag', { [`nhsuk-tag--${color}`]: color }, className)}
    {...rest}
  />
);

export default Tag;
