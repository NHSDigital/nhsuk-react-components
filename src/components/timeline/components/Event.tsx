import React, { ReactNode } from 'react';

export interface EventProps {
  title: ReactNode;
  instigator?: ReactNode;
  date: string;
  description?: Array<ReactNode>;
  action?: ReactNode;
}

const Event: React.FC<EventProps> = ({ title, instigator, date, description = [], action }) => (
  <div className="nhsuk-timeline__event nhsuk-u-margin-bottom-2">
    <h2 className="nhsuk-timeline__title">
      <span className="nhsuk-timeline__status">{title}</span>
      {instigator && <span className="nhsuk-timeline__by"> by {instigator}</span>}
      {action && <span className="nhsuk-timeline__action">{action}</span>}
    </h2>
    <p className="nhsuk-timeline__date">
      <time dateTime={date}>{date}</time>
    </p>
    {description.map((descriptionItem, index) => (
      <p key={index.toString()} className="nhsuk-timeline__description-item">
        {descriptionItem}
      </p>
    ))}
  </div>
);

export default Event;
