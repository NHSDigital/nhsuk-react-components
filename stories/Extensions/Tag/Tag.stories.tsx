import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import Tag from '@components/content-presentation/tag/Tag';

const meta: Meta<typeof Tag> = {
  title: 'Extensions/Tags',
  component: Tag,
};
export default meta;

type Story = StoryObj<typeof Tag>;

export const Basic: Story = {
  render: () => (
    <div className="tag-demo">
      <Tag>Standard</Tag>
      <Tag>Done</Tag>
      <Tag color="white">Started</Tag>
      <Tag color="grey">Not Started</Tag>
      <Tag color="blue">Ready to submit</Tag>
      <Tag color="red">FP69</Tag>
      <Tag color="orange">Ceased - no cervix</Tag>
    </div>
  ),
};

export const Colours: Story = {
  render: () => (
    <div className="tag-demo">
      <Tag>Standard</Tag>
      <Tag color="white">Started</Tag>
      <Tag color="grey">Not Started</Tag>
      <Tag color="green">New</Tag>
      <Tag color="aqua-green">Active</Tag>
      <Tag color="blue">Pending</Tag>
      <Tag color="purple">Received</Tag>
      <Tag color="pink">Sent</Tag>
      <Tag color="red">Rejected</Tag>
      <Tag color="orange">Declined</Tag>
      <Tag color="yellow">Delayed</Tag>
    </div>
  ),
};
