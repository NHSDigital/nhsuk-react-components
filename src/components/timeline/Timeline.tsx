import React from 'react';
import Event, { EventProps } from './components/Event';

type TimelineProps = {
  events: Array<EventProps>;
};

const Timeline: React.FC<TimelineProps> = ({ events }) => (
  <div className="nhsuk-timeline">
    {events.map(({ title, instigator, date, description, action }, index) => (
      <Event
        key={index.toString()}
        title={title}
        instigator={instigator}
        date={date}
        description={description}
        action={action}
      />
    ))}
  </div>
);

export default Timeline;
