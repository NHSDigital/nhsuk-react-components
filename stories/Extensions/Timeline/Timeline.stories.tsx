/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import Timeline from '@components/timeline';
import { EventProps } from '@components/timeline/components/Event';
import Tag from '@components/content-presentation/tag/Tag';

const meta: Meta<typeof Timeline> = {
  title: 'Extensions/Timeline',
  component: Timeline,
};
export default meta;

type Story = StoryObj<typeof Timeline>;

const actionStyle = {
  marginLeft: '8px',
  color: '#768692',
};
const tagStyle = {
  marginLeft: '8px',
};
const linkStyle = {
  marginLeft: '8px',
  fontWeight: 400,
};

const storybookEvents: EventProps[] = [
  {
    title: <>Result sent</>,
    instigator: <>System</>,
    date: '19 Nov 2019, 7:49:10 pm',
    description: [
      <>Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised</>,
      <>Test Date: 19-Oct-2020, 9:00:00 am</>,
    ],
    action: (
      <span style={actionStyle}>
        -
        <Tag style={tagStyle} color="yellow">
          Send to printer
        </Tag>
        <a style={linkStyle} href="/placeholder/">
          Cancel
        </a>
      </span>
    ),
  },
  {
    title: <>Patient deferred</>,
    instigator: <>James Smith</>,
    date: '19 Nov 2019, 4:28:57 pm',
    description: [
      <>Defer Reason: Pregnancy</>,
      <>CRM Case Number: CAS-12345-ABCDE</>,
      <>Comments: Requested via the GP form</>,
    ],
  },
  {
    title: <>Next test due date changed</>,
    instigator: <>System</>,
    date: '19 Nov 2019, 11:59:59 am',
    description: [<>Next test due date changed to 10-Oct-2021</>],
  },
  {
    title: <>Test result added</>,
    date: '19 Nov 2019, 9:12:42 am',
    description: [<>Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised</>],
  },
  {
    title: <>Screening invitation sent</>,
    date: '18 Nov 2019, 7:49:10 pm',
    action: (
      <span style={actionStyle}>
        -
        <Tag style={tagStyle} color="grey">
          Sent to patient
        </Tag>
        <a style={linkStyle} href="/placeholder2">
          Resend
        </a>
      </span>
    ),
  },
];

export const Standard: Story = {
  render: () => <Timeline events={storybookEvents} />,
};
