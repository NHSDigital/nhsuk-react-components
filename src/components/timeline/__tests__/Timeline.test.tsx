import React from 'react';
import Timeline from '../Timeline';
import {render} from '@testing-library/react'
import Event, { EventProps } from '../components/Event';
import Tag from '../../tag/Tag';

const actionLinkText = 'Cancel';
const eventInstigator = 'System';

const testEvents: EventProps[] = [
  {
    title: <>Result sent</>,
    instigator: <>{eventInstigator}</>,
    date: '19 Nov 2019, 7:49:10 pm',
    description: [
      <>Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised</>,
      <>Test Date: 19-Oct-2020, 9:00:00 am</>,
    ],
    action: (
      <span className="nhsuk-timeline__action">
        -
        <Tag className="nhsuk-timeline__tag" color="yellow">
          Send to printer
        </Tag>
        <a className="nhsuk-timeline__link" href="/placeholder/">
          Cancel
        </a>
      </span>
    ),
  },
  {
    title: <>Patient deferred</>,
    date: '19 Nov 2019, 4:28:57 pm',
    description: [
      <>Defer Reason: Pregnancy</>,
      <>CRM Case Number: CAS-12345-ABCDE</>,
      <>Comments: Requested via the GP form</>,
    ],
  },
  {
    title: <>Screening invitation sent</>,
    instigator: <>System</>,
    date: '18 Nov 2019, 7:49:10 pm',
    action: (
      <span className="nhsuk-timeline__action">
        -
        <Tag className="nhsuk-timeline__tag" color="grey">
          Sent to patient
        </Tag>
        <a className="nhsuk-timeline__link" href="/placeholder2">
          Resend
        </a>
      </span>
    ),
  },
];

describe('Timeline', () => {
  it('matches snapshot', () => {
    const timeline = render(<Timeline events={testEvents} />);

    expect(timeline).toMatchSnapshot();

  });

  it('should have correct classes applied', () => {
    const timeline = render(<Timeline events={testEvents} />);

    expect(timeline.container.querySelector('.nhsuk-timeline')).toBeTruthy();
  });

  describe('Event', () => {
    it('matches snapshot', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event).toMatchSnapshot();

    });

    it('should have correct classes applied', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event.container.querySelector('.nhsuk-timeline__event')).toBeTruthy();

    });

    it('should have correct title', () => {
      const event = render(<Event {...testEvents[1]} />);

      expect(event.container.querySelector('.nhsuk-timeline__title')?.textContent).toBe('Patient deferred');

    });

    it('should have correct instigator if instigator data is present', () => {
      const event = render(<Event {...testEvents[0]} />);

     expect(event.container.querySelector('.nhsuk-timeline__by')).toBeTruthy();
      expect(event.container.querySelector('.nhsuk-timeline__by')?.textContent).toBe(` by ${eventInstigator}`);

    });

    it('should not have instigator if instigator data is not present', () => {
      const event = render(<Event {...testEvents[1]} />);

      expect(event.container.querySelector('.nhsuk-timeline__by')).toBeFalsy();

    });

    it('should have correct date', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event.container.querySelector('.nhsuk-timeline__date time')?.textContent).toBe(testEvents[0].date);
    });

    it('has correct description items if description data is present', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event.container.querySelector('.nhsuk-timeline__description-item')).toBeTruthy();
    });

    it('has no description items if description data is not present', () => {
      const event = render(<Event {...testEvents[2]} />);

      expect(event.container.querySelector('.nhsuk-timeline__description-item')).toBeFalsy();
    });

    it('has correct link text if action data is present', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event.container.querySelector('.nhsuk-timeline__link')?.textContent).toBe(actionLinkText);
    });

    it('will show action if action data present', () => {
      const event = render(<Event {...testEvents[0]} />);

      expect(event.container.querySelector('.nhsuk-timeline__action')).toBeTruthy();
    });

    it('will not show action if no action data present', () => {
      const event = render(<Event {...testEvents[1]} />);

      expect(event.container.querySelector('.nhsuk-timeline__action')).toBeFalsy();
    });
  });
});
