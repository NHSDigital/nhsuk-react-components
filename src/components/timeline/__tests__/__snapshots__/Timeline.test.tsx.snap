// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Event matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
      >
        <h2
          class="nhsuk-timeline__title"
        >
          <span
            class="nhsuk-timeline__status"
          >
            Result sent
          </span>
          <span
            class="nhsuk-timeline__by"
          >
             by 
            System
          </span>
          <span
            class="nhsuk-timeline__action"
          >
            <span
              class="nhsuk-timeline__action"
            >
              -
              <span
                class="nhsuk-tag nhsuk-tag--yellow nhsuk-timeline__tag"
              >
                Send to printer
              </span>
              <a
                class="nhsuk-timeline__link"
                href="/placeholder/"
              >
                Cancel
              </a>
            </span>
          </span>
        </h2>
        <p
          class="nhsuk-timeline__date"
        >
          <time
            datetime="19 Nov 2019, 7:49:10 pm"
          >
            19 Nov 2019, 7:49:10 pm
          </time>
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Test Date: 19-Oct-2020, 9:00:00 am
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
    >
      <h2
        class="nhsuk-timeline__title"
      >
        <span
          class="nhsuk-timeline__status"
        >
          Result sent
        </span>
        <span
          class="nhsuk-timeline__by"
        >
           by 
          System
        </span>
        <span
          class="nhsuk-timeline__action"
        >
          <span
            class="nhsuk-timeline__action"
          >
            -
            <span
              class="nhsuk-tag nhsuk-tag--yellow nhsuk-timeline__tag"
            >
              Send to printer
            </span>
            <a
              class="nhsuk-timeline__link"
              href="/placeholder/"
            >
              Cancel
            </a>
          </span>
        </span>
      </h2>
      <p
        class="nhsuk-timeline__date"
      >
        <time
          datetime="19 Nov 2019, 7:49:10 pm"
        >
          19 Nov 2019, 7:49:10 pm
        </time>
      </p>
      <p
        class="nhsuk-timeline__description-item"
      >
        Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised
      </p>
      <p
        class="nhsuk-timeline__description-item"
      >
        Test Date: 19-Oct-2020, 9:00:00 am
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Timeline matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nhsuk-timeline"
      >
        <div
          class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
        >
          <h2
            class="nhsuk-timeline__title"
          >
            <span
              class="nhsuk-timeline__status"
            >
              Result sent
            </span>
            <span
              class="nhsuk-timeline__by"
            >
               by 
              System
            </span>
            <span
              class="nhsuk-timeline__action"
            >
              <span
                class="nhsuk-timeline__action"
              >
                -
                <span
                  class="nhsuk-tag nhsuk-tag--yellow nhsuk-timeline__tag"
                >
                  Send to printer
                </span>
                <a
                  class="nhsuk-timeline__link"
                  href="/placeholder/"
                >
                  Cancel
                </a>
              </span>
            </span>
          </h2>
          <p
            class="nhsuk-timeline__date"
          >
            <time
              datetime="19 Nov 2019, 7:49:10 pm"
            >
              19 Nov 2019, 7:49:10 pm
            </time>
          </p>
          <p
            class="nhsuk-timeline__description-item"
          >
            Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised
          </p>
          <p
            class="nhsuk-timeline__description-item"
          >
            Test Date: 19-Oct-2020, 9:00:00 am
          </p>
        </div>
        <div
          class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
        >
          <h2
            class="nhsuk-timeline__title"
          >
            <span
              class="nhsuk-timeline__status"
            >
              Patient deferred
            </span>
          </h2>
          <p
            class="nhsuk-timeline__date"
          >
            <time
              datetime="19 Nov 2019, 4:28:57 pm"
            >
              19 Nov 2019, 4:28:57 pm
            </time>
          </p>
          <p
            class="nhsuk-timeline__description-item"
          >
            Defer Reason: Pregnancy
          </p>
          <p
            class="nhsuk-timeline__description-item"
          >
            CRM Case Number: CAS-12345-ABCDE
          </p>
          <p
            class="nhsuk-timeline__description-item"
          >
            Comments: Requested via the GP form
          </p>
        </div>
        <div
          class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
        >
          <h2
            class="nhsuk-timeline__title"
          >
            <span
              class="nhsuk-timeline__status"
            >
              Screening invitation sent
            </span>
            <span
              class="nhsuk-timeline__by"
            >
               by 
              System
            </span>
            <span
              class="nhsuk-timeline__action"
            >
              <span
                class="nhsuk-timeline__action"
              >
                -
                <span
                  class="nhsuk-tag nhsuk-tag--grey nhsuk-timeline__tag"
                >
                  Sent to patient
                </span>
                <a
                  class="nhsuk-timeline__link"
                  href="/placeholder2"
                >
                  Resend
                </a>
              </span>
            </span>
          </h2>
          <p
            class="nhsuk-timeline__date"
          >
            <time
              datetime="18 Nov 2019, 7:49:10 pm"
            >
              18 Nov 2019, 7:49:10 pm
            </time>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nhsuk-timeline"
    >
      <div
        class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
      >
        <h2
          class="nhsuk-timeline__title"
        >
          <span
            class="nhsuk-timeline__status"
          >
            Result sent
          </span>
          <span
            class="nhsuk-timeline__by"
          >
             by 
            System
          </span>
          <span
            class="nhsuk-timeline__action"
          >
            <span
              class="nhsuk-timeline__action"
            >
              -
              <span
                class="nhsuk-tag nhsuk-tag--yellow nhsuk-timeline__tag"
              >
                Send to printer
              </span>
              <a
                class="nhsuk-timeline__link"
                href="/placeholder/"
              >
                Cancel
              </a>
            </span>
          </span>
        </h2>
        <p
          class="nhsuk-timeline__date"
        >
          <time
            datetime="19 Nov 2019, 7:49:10 pm"
          >
            19 Nov 2019, 7:49:10 pm
          </time>
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Test Result: (39S) Low-grade dyskaryosis, HPV positive, Repeat Advised
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Test Date: 19-Oct-2020, 9:00:00 am
        </p>
      </div>
      <div
        class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
      >
        <h2
          class="nhsuk-timeline__title"
        >
          <span
            class="nhsuk-timeline__status"
          >
            Patient deferred
          </span>
        </h2>
        <p
          class="nhsuk-timeline__date"
        >
          <time
            datetime="19 Nov 2019, 4:28:57 pm"
          >
            19 Nov 2019, 4:28:57 pm
          </time>
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Defer Reason: Pregnancy
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          CRM Case Number: CAS-12345-ABCDE
        </p>
        <p
          class="nhsuk-timeline__description-item"
        >
          Comments: Requested via the GP form
        </p>
      </div>
      <div
        class="nhsuk-timeline__event nhsuk-u-margin-bottom-2"
      >
        <h2
          class="nhsuk-timeline__title"
        >
          <span
            class="nhsuk-timeline__status"
          >
            Screening invitation sent
          </span>
          <span
            class="nhsuk-timeline__by"
          >
             by 
            System
          </span>
          <span
            class="nhsuk-timeline__action"
          >
            <span
              class="nhsuk-timeline__action"
            >
              -
              <span
                class="nhsuk-tag nhsuk-tag--grey nhsuk-timeline__tag"
              >
                Sent to patient
              </span>
              <a
                class="nhsuk-timeline__link"
                href="/placeholder2"
              >
                Resend
              </a>
            </span>
          </span>
        </h2>
        <p
          class="nhsuk-timeline__date"
        >
          <time
            datetime="18 Nov 2019, 7:49:10 pm"
          >
            18 Nov 2019, 7:49:10 pm
          </time>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
